packages/@react-spectrum/numberfield/docs/NumberField.mdx

```diff
@@ -0,0 +1,218 @@
+
+import {Layout} from '@react-spectrum/docs';
+export default Layout;
+
+import docs from 'docs:@react-spectrum/numberfield';
+import {HeaderInfo, PropTable} from '@react-spectrum/docs';
+import packageData from '@react-spectrum/numberfield/package.json';
+
+```jsx import
+import {NumberField} from '@react-spectrum/numberfield';
+import {Flex} from '@react-spectrum/layout';
+```
+
+---
+category: Forms
+keywords: [number field, input]
+---
+
+# NumberField
+
+<p>{docs.exports.NumberField.description}</p>
+
+<HeaderInfo
+  packageData={packageData}
+  componentNames={['NumberField']}
+  sourceData={[]} />
+
+## Example
+
+```tsx example
+<NumberField label="Name" />
+```
+
+## Value
+
+A NumberField's `value` is empty by default, but an initial, uncontrolled, value can be provided using the `defaultValue` prop.
+Alternatively, a controlled value can be provided using the `value` prop.
+
+```tsx example
+function Example() {
+  let [value, setValue] = React.useState(15);
+
+  return (
+    <Flex gap="size-150" wrap>
+      <NumberField
+        label="Cookies (Uncontrolled)"
+        defaultValue={15} />
+
+      <NumberField
+        label="Cookies (Controlled)"
+        value={value}
+        onChange={setValue} />
+    </Flex>
+  );
+}
+```
+
+## Steps
+
+A NumberField can have a specified step size, as well as a min and max value. The steps will always start at the `minValue` or `0`
+and will not exceed the min or the max if they are specified.
+
+```tsx example
+<NumberField label="Cookies" step={3} minValue={-5} maxValue={21} />
+```
+
+## Labeling
+
+A visual label should be provided for the NumberField using the `label` prop. If the NumberField is required, the `isRequired` and
+`necessityIndicator` props can be used to show a required state.
+
+```tsx example
+<Flex gap="size-150" wrap>
+  <NumberField label="Cookies" />
+  <NumberField label="Cookies" isRequired necessityIndicator="icon" />
+  <NumberField label="Cookies" isRequired necessityIndicator="label" />
+  <NumberField label="Cookies" necessityIndicator="label" />
+</Flex>
+```
+
+### Accessibility
+
+If a visible label isn't specified, an `aria-label` must be provided to the NumberField for
+accessibility. If the field is labeled by a separate element, an `aria-labelledby` prop must be provided using
+the `id` of the labeling element instead.
+
+### Internationalization
+
+#### Label
+In order to internationalize a NumberField, a localized string should be passed to the `label` or `aria-label` prop.
+When the `necessityIndicator` prop is set to `"label"`, a localized string will be provided for `"(required)"` or `"(optional)"` automatically.
+
+#### Value
+The NumberField value can be formatted by using the `formatOptions` prop.
+`formatOptions` is compatible with the option parameter of [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat) and is applied based on the current locale.
```

Comment:
I think it may be useful to explain each of the formatting styles in more detail and call out their nuances. Maybe reuse the decimals, percentages, currency values, and units sections from the aria docs as sub-sections? https://github.com/adobe/react-spectrum/pull/1523/files#diff-baa7d39d9b1465a7d8ce5b1760066982303b4f8a93fc797af787879bef41bfb4R170-R252